src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/poptimizer/cardinality/VarSpecificCardinalityEstimationImpl.java

```diff
@@ -95,6 +88,13 @@ else if ( rootOp instanceof LogicalOpJoin ) {
 		else if ( rootOp instanceof LogicalOpUnion ) {
 			return _initiateUnionCardinalityEstimation( plan.getSubPlan(0), plan.getSubPlan(1), v );
 		}
+		else if ( rootOp instanceof LogicalOpMultiwayUnion ) {
+			final List<PhysicalPlan> subPlans = new ArrayList<>();
+			for( int i = 0; i < plan.numberOfSubPlans(); i++ ) {
+				subPlans.add( plan.getSubPlan(i) );
+			}
```

Comment:
I suggest to avoid creating and populating this extra list by, instead, simply passing the `plan` directly to the helper function `_initiateMultiwayUnionCardinalityEstimation`. The for-loops in that function can then iterate over the subplans in the way you do it here (i.e., based on `int i` and calling `plan.getSubPlan(i)`).